build_test.go

```diff
@@ -369,6 +378,37 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 			Expect(dotnetSymlinker.LinkCall.Receives.LayerPath).To(Equal(filepath.Join(layersDir, "dotnet-core-sdk")))
 		})
 	})
+
+	context("when the sdk version is set via buildpack.yml", func() {
```

Comment:
This is kind of weird to me because in the top level `entryResolver` fake (line 60) you have:
```
	entryResolver.ResolveCall.Returns.BuildpackPlanEntry = packit.BuildpackPlanEntry{
			Name: "dotnet-sdk",
			Metadata: map[string]interface{}{
				"version-source": "buildpack.yml",
				"version":        "2.5.x",
				"build":          true,
				"launch":         true,
			},
		}
```
so this test isn't really different than any of the others. I think it would be best if just this test used the `buildpack.yml`, and then the top level `entryResolver` fake had a different version source. That way, when we come in and deprecate `buildpack.yml`, we only need to remove one test, rather than change every single one.